isis/tests/FunctionalTestsKaguyami2isis.cpp

```diff
@@ -0,0 +1,388 @@
+  Pvl *isisLabel = cube.label();
+
+  // Dimensions group
+  EXPECT_EQ(cube.sampleCount(), 320);
+  EXPECT_EQ(cube.lineCount(), 20);
+  EXPECT_EQ(cube.bandCount(), 2);
+
+  // Pixels group
+  EXPECT_EQ(PixelTypeName(cube.pixelType()), "SignedWord");
+  EXPECT_EQ(ByteOrderName(cube.byteOrder()), "Lsb");
+  EXPECT_DOUBLE_EQ(cube.base(), 0.0);
+  EXPECT_DOUBLE_EQ(cube.multiplier(), 0.013);
+
+  // Instrument Group
+  PvlGroup &inst = isisLabel->findGroup("Instrument", Pvl::Traverse);
+  EXPECT_EQ(inst["MissionName"][0].toStdString(), "SELENE");
+  EXPECT_EQ(inst["SpacecraftName"][0].toStdString(), "KAGUYA");
+  EXPECT_EQ(inst["InstrumentName"][0].toStdString(), "Multiband Imager Near Infrared");
+  EXPECT_EQ(inst["InstrumentId"][0].toStdString(), "MI-NIR");
+  EXPECT_EQ(inst["TargetName"][0].toStdString(), "MOON");
+  EXPECT_EQ(inst["StartTime"][0].toStdString(), "2008-09-16T20:10:30.480257");
+  EXPECT_EQ(inst["StopTime"][0].toStdString(), "2008-09-16T20:10:42.921232");
+  EXPECT_EQ(inst["SpacecraftClockStartCount"][0].toStdString(), "905631021.132");
+  EXPECT_EQ(inst["SpacecraftClockStopCount"][0].toStdString(), "905631033.574");
+  EXPECT_DOUBLE_EQ(inst["LineExposureDuration"], 13.2148);
+  EXPECT_EQ(inst["LineExposureDuration"].unit(), "msec");
+  EXPECT_DOUBLE_EQ(inst["LineSamplingInterval"], 39);
+  EXPECT_EQ(inst["LineSamplingInterval"].unit(), "msec");
+  EXPECT_DOUBLE_EQ(inst["CorrectedSamplingInterval"], 38.999922);
+  EXPECT_EQ(inst["CorrectedSamplingInterval"].unit(), "msec");
+
+  // Archive Group
+  PvlGroup &arch = isisLabel->findGroup("Archive", Pvl::Traverse);
+
+  EXPECT_EQ(arch["DataSetId"][0].toStdString(), "MI-NIR_Level2B");
+  EXPECT_EQ(arch["ProductSetId"][0].toStdString(), "MI-NIR_Level2B2");
+
+  // Bandbin Group
+  PvlGroup &bandBin = isisLabel->findGroup("BandBin", Pvl::Traverse);
+  std::istringstream bandBinStream(R"(
+  Group = BandBin
+    FilterName = (MV1, MV2, MV3, MV4, MV5)
+    Center     = (414.0, 749.0, 901.0, 950.0, 1001.0) <nm>
+    Width      = (20.0, 12.0, 21.0, 30.0, 42.0) <nm>
+    BaseBand   = MV5
+  End_Group
+  )");
+  PvlGroup bandBinTruth;
+  bandBinStream >> bandBinTruth;
+  AssertPvlGroupEqual("resultingBandBin", "truthBandBin", bandBin, bandBinTruth);
+
+  // Kernels Group
+  PvlGroup &kern = isisLabel->findGroup("Kernels", Pvl::Traverse);
+  EXPECT_EQ(int(kern["NaifCkCode"]), -131340);
+  EXPECT_EQ(int(kern["NaifFrameCode"]), -131341);
+
+  std::unique_ptr<Histogram> hist (cube.histogram());
+
+  EXPECT_DOUBLE_EQ(hist->Average(), 29.395262812500022);
+  EXPECT_DOUBLE_EQ(hist->Sum(), 188129.68200000015);
+  EXPECT_EQ(hist->ValidPixels(), 6400);
+  EXPECT_DOUBLE_EQ(hist->StandardDeviation(), 2.8449125231835715);
+}
+
+TEST(kaguyatc2isisTest, FunctionalTestsKaguyami2isisProj) {
+  QTemporaryDir prefix;
+  QString cubeFileName = prefix.path() + "/kaguyatc2isisTEMP.cub";
+  QVector<QString> args = {"from=data/kaguyami2isis/MVA_2B2_01_01229N603E1343_cropped", "to="+cubeFileName};
```

Comment:
the Vis tests is also a Vis 2b2 image, is this test redundant or am I missing something?

Reply:
No, that is a fair point. The original default test for the mi camera tested three separate images and this was one of them. To be honest I am not sure why this image was special compared to the NIR and VIS images that were also being tested.

Reply:
It seems like the projected test is supposed to be 3C5 file, and it is supposed to fail as an unsupported file type. It seems like the thing to do is replace this with a failure check without a cropped image and instead write a PVL blob to disk with a PVL group named `IMAGE_MAP_PROJECTION` to trigger the error.

Reply:
I pulled the label from the projected cube that was in the projected test. Since it's a pds label and not an ISIS label I am not sure on the mechanics on adding the label in test rather than just using an existing label. I could spend some time trying to figure that out but this seemed like the faster solution

Reply:
that's fine enough, I think the thing to do was create a PVL object from a string literal and write that to disk in the test itself.